File: yggdrasil/examples/hello/src/hello.m
disp('Hello from Matlab');

% Ins/outs matching with the the model yaml
inf = YggInterface('YggInput', 'inFile');
outf = YggInterface('YggOutput', 'outFile');
inq = YggInterface('YggInput', 'helloQueueIn');
outq = YggInterface('YggOutput', 'helloQueueOut');
disp('hello(M): Created I/O channels');

% Receive input from a local file
[flag, buf] = inf.recv();
if (~flag);
  error('hello(M): ERROR FILE RECV');
end
fprintf('hello(M): Received %d bytes from file: %s\n', ...
	length(buf), buf);

% Send output to the output queue
ret = outq.send(buf);
if (~ret);
  error('hello(M): ERROR QUEUE SEND');
end
disp('hello(M): Sent to outq');
outq.send_eof();

% Receive input form the input queue
[flag, buf] = inq.recv();
if (~flag);
  error('hello(M): ERROR QUEUE RECV');
end
fprintf('hello(M): Received %d bytes from queue: %s\n', ...
	length(buf), buf);

% Send output to a local file
ret = outf.send(buf);
if (~ret);
  error('hello(M): ERROR FILE SEND');
end
disp('hello(M): Sent to outf');

disp('Goodbye from Matlab');

File: yggdrasil/examples/hello/src/hello.py
from __future__ import print_function
from yggdrasil.interface.YggInterface import YggInput, YggOutput


def runhello():
    print('Hello from Python')

    # Ins/outs matching with the the model yaml
    inf = YggInput('inFile')
    outf = YggOutput('outFile')
    inq = YggInput('helloQueueIn')
    outq = YggOutput('helloQueueOut')
    print("hello(P): Created I/O channels")

    # Receive input from a local file
    ret, buf = inf.recv()
    if not ret:
        raise RuntimeError('hello(P): ERROR FILE RECV')
    print('hello(P): Received %d bytes from file: %s' % (len(buf), buf))

    # Send output to the output queue
    ret = outq.send(buf)
    if not ret:
        raise RuntimeError('hello(P): ERROR QUEUE SEND')
    print('hello(P): Sent to outq')

    # Receive input form the input queue
    ret, buf = inq.recv()
    if not ret:
        raise RuntimeError('hello(P): ERROR QUEUE RECV')
    print('hello(P): Received %d bytes from queue: %s' % (len(buf), buf))

    # Send output to a local file
    ret = outf.send(buf)
    if not ret:
        raise RuntimeError('hello(P): ERROR FILE SEND')
    print('hello(P): Sent to outf')

    print('Goodbye from Python')

    
if __name__ == '__main__':
    runhello()
